app/controllers/katello/products_controller.rb

```diff
@@ -30,6 +30,27 @@ def available_repositories
             end
           end
 
+          @product.repositories.each do |product_repo|
+            found = repos.detect do |repo|
+              product_repo.pulp_id == repo[:pulp_id]
+            end
+            unless found
+              repos << {
+                  :repo_name => product_repo.name,
+                  :path => product_repo.url,
+                  :pulp_id => product_repo.pulp_id,
+                  :content_id => product_repo.content_id,
+                  :substitutions => {
+                      :basearch => product_repo.arch,
+                      :releasever => product_repo.minor
+                  },
+                  :enabled => true,
+                  :promoted => product_repo.promoted?,
+                  :registry_name => product_repo.docker_upstream_name
+              }
+            end
```

Comment:
This logic is a little hard to follow with its intent. Breaking the logic chunks into methods with names that convey intent might help.